spring-integration-mail/src/test/java/org/springframework/integration/mail/TestMailServer.java

```diff
@@ -282,10 +282,10 @@ else if (line.contains("FETCH 1 (ENVELOPE")) {
 							write("* 1 FETCH (RFC822.SIZE 6909 INTERNALDATE \"27-May-2013 09:45:41 +0000\" "
 									+ "FLAGS (\\Seen) "
 									+ "ENVELOPE (\"Mon, 27 May 2013 15:14:49 +0530\" "
-									+ "\"Test Email\" ((\"Foo\" NIL \"foo\" \"bar.tv\")) "
+									+ "\"Test Email\" ((\"Bar\" NIL \"bar\" \"baz\")) "
 									+ "((\"Foo\" NIL \"foo\" \"bar.tv\")) "
 									+ "((\"Foo\" NIL \"foo\" \"bar.tv\")) "
-									+ "((\"Bar\" NIL \"bar\" \"baz.net\")) NIL NIL "
+									+ "((\"Foo\" NIL \"foo\" \"bar\")) NIL NIL "
```

Comment:
Wow! That was so fast :smile: 

So, this `ENVELOPE` definition should correlate with the `BODY[]`?
Or what's going on?

Thanks!
Will merge after explanation and apply the same fix to the DSL PR: https://github.com/spring-projects/spring-integration-java-dsl/pull/103